webapp/client/src/pages/TaxNumberPage.js

```diff
@@ -10,36 +10,178 @@ import FormRowCentered from "../components/FormRowCentered";
+        showTaxNumberExists: true,
+        showBundesland: true,
+        showBufaNr: true,
+        showSteuernummer: false,
+        showRequestNewTaxNumber: false,
+      };
+    case TAX_NUMBER_FORM_STATE_CHANGES.TaxOfficeSelected:
+      return {
+        ...state,
+        bufaNr: action.value,
+        requestNewTaxNumber: undefined,
+        showTaxNumberExists: true,
+        showBundesland: true,
+        showBufaNr: true,
+        showSteuernummer: false,
+        showRequestNewTaxNumber: true,
+      };
+    default:
+      return {
+        ...state,
+        taxNumberExists: undefined,
+        bundesland: undefined,
+        bufaNr: undefined,
+        steuernummer: [""],
+        requestNewTaxNumber: undefined,
+        showTaxNumberExists: true,
+        showBundesland: false,
+        showBufaNr: false,
+        showSteuernummer: false,
+        showRequestNewTaxNumber: false,
+      };
+  }
+}
+
+function getSplitTypeForState(selectedStateAbbreviation) {
+  switch (selectedStateAbbreviation) {
+    case "bw":
+    case "be":
+    case "hb":
+    case "hh":
+    case "nd":
+    case "rp":
+    case "sh":
+      return "0";
+    case "by":
+    case "bb":
+    case "mv":
+    case "sl":
+    case "sn":
+    case "st":
+    case "th":
+      return "1";
+    case "nw":
+      return "2";
+    default:
+      return "3";
+  }
+}
+
+function getInitialTaxNumberPageData(fields) {
+  const currentState = {
+    taxNumberExists: fields.steuernummerExists.value,
+    bundesland: fields.bundesland.selectedValue,
+    bufaNr: fields.bufaNr.selectedValue,
+    steuernummer: fields.steuernummer.value,
+    requestNewTaxNumber: fields.requestNewTaxNumber.checked,
+    showTaxNumberExists: true,
+    showBundesland: false,
+    showBufaNr: false,
+    showSteuernummer: false,
+    showRequestNewTaxNumber: false,
+  };
+  if (fields.steuernummerExists.value === "yes") {
+    if (fields.bundesland.selectedValue) {
+      return {
+        ...currentState,
+        showBundesland: true,
+        showSteuernummer: true,
+      };
     }
-    if (fields.steuernummerExists.value === "no") {
-      if (fields.bundesland.selectedValue) {
-        if (fields.bufaNr.selectedValue) {
-          return TAX_NUMBER_FORM_STATES.RequestNewTaxNumber;
-        }
-        return TAX_NUMBER_FORM_STATES.TaxOfficeSelection;
+    return {
+      ...currentState,
+      showBundesland: true,
+    };
+  }
+  if (fields.steuernummerExists.value === "no") {
+    if (fields.bundesland.selectedValue) {
+      if (fields.bufaNr.selectedValue) {
+        return {
+          ...currentState,
+          showBundesland: true,
+          showBufaNr: true,
+          showRequestNewTaxNumber: true,
+        };
       }
-      return TAX_NUMBER_FORM_STATES.StateSelectionNoTaxNumberExists;
+      return {
+        ...currentState,
+        showBundesland: true,
+        showBufaNr: true,
+      };
     }
+    return {
+      ...currentState,
+      showBundesland: true,
+    };
```

Comment:
Can't we spare all this and instead do something like this?

```
showTaxNumberExists: true,
showBundesland: bool(fields.steuernummerExists.value),
showBufaNr: fields.steuernummerExists.value === "no" && fields.steuernummerExists.value,
...
```

Reply:
Great idea